client/authority-discovery/src/worker/tests.rs

```diff
@@ -299,63 +301,62 @@ fn publish_discover_cycle() {
 	};
 
 	let key_store = KeyStore::new();
-	let node_a_public = key_store
-		.write()
-		.sr25519_generate_new(key_types::AUTHORITY_DISCOVERY, None)
-		.unwrap();
-	let test_api = Arc::new(TestApi {
-		authorities: vec![node_a_public.into()],
-	});
-
-	let (_to_worker, from_service) = mpsc::channel(0);
-	let mut worker = Worker::new(
-		from_service,
-		test_api,
-		network.clone(),
-		vec![],
-		dht_event_rx.boxed(),
-		Role::Authority(key_store),
-		None,
-	);
-
-	worker.publish_ext_addresses().unwrap();
-
-	// Expect authority discovery to put a new record onto the dht.
-	assert_eq!(network.put_value_call.lock().unwrap().len(), 1);
-
-	let dht_event = {
-		let (key, value) = network.put_value_call.lock().unwrap().pop().unwrap();
-		sc_network::DhtEvent::ValueFound(vec![(key, value)])
-	};
 
-	// Node B discovering node A's address.
-
-	let (mut dht_event_tx, dht_event_rx) = channel(1000);
-	let test_api = Arc::new(TestApi {
-		// Make sure node B identifies node A as an authority.
-		authorities: vec![node_a_public.into()],
-	});
-	let network: Arc<TestNetwork> = Arc::new(Default::default());
-	let key_store = KeyStore::new();
+	let _ = pool.spawner().spawn_local_obj(async move {
```

Comment:
Is this pool ever executing?

Reply:
Good that you brought this up. If i call `pool.run()` at the end so that i drive all the spawned tasks to completion, i see the error:
```
thread 'worker::tests::publish_discover_cycle' panicked at 'cannot execute `LocalPool` executor from within another executor: EnterError', /home/rakan/.cargo/registry/src/github.com-1ecc6299db9ec823/futures-executor-0.3.5/src/local_pool.rs:78:26
```
Though i don't have any other pool running. What is this other executor that is running in this case?